packages/trpc/server/routers/viewer/organizations/create.handler.ts

```diff
@@ -82,6 +82,8 @@ export const createHandler = async ({ input, ctx }: CreateOptions) => {
   const availability = getAvailabilityFromSchedule(DEFAULT_SCHEDULE);
 
   if (check === false) {
+    if (IS_CALCOM) await vercelCreateDomain(slug);
```

Comment:
Moved up to fail before creating the team if vercel domain creation fails. And relying on whether the environment is either Dev, QA, or PROD